src/render/draw_globe_atmosphere.js

```diff
@@ -19,26 +20,37 @@ function drawGlobeAtmosphere(painter: Painter) {
     const depthMode = new DepthMode(gl.LEQUAL, DepthMode.ReadOnly, [0, 1]);
     const program = painter.useProgram('globeAtmosphere');
 
-    // Compute center and approximate radius of the globe on screen coordinates
+    // Render the gradient atmosphere by casting rays from screen pixels and determining their
+    // closest distance to the globe. This is done in view space where camera is located in the origo
+    // facing -z direction.
     const viewMatrix = transform._camera.getWorldToCamera(transform.worldSize, 1.0);
     const viewToProj = transform._camera.getCameraToClipPerspective(transform._fov, transform.width / transform.height, transform._nearZ, transform._farZ);
-    const globeToView = mat4.mul([], viewMatrix, transform.globeMatrix);
-    const viewToScreen = mat4.mul([], transform.labelPlaneMatrix, viewToProj);
+    const projToView = mat4.invert([], viewToProj);
 
-    const centerOnViewSpace = vec3.transformMat4([], [0, 0, 0], globeToView);
-    const radiusOnViewSpace = vec3.add([], centerOnViewSpace, [transform.worldSize / Math.PI / 2.0, 0, 0]);
+    const project = (point: Vec3, m: Mat4): [number, number, number] => {
+        vec3.transformMat4(point, point, m);
+        return [point[0], point[1], point[2]];
```

Comment:
nit: We should be able to directly `return vec3.transformMat4(point, point, m);` here, it mutates the input `point` variable anyway, so I don't think we need the extra array:
```suggestion
        return vec3.transformMat4(point, point, m);
```

Reply:
I actually spent way more time on this function than I wanted to. The problem here is that `vec3.transformMat4()` expects the input coordinate to have a type of `Vec3` which is defined as `type VecType = Array<number> | Float32Array | Float64Array;`. This type will eventually need to be converted/casted to a tuple in order to be passed to shaders as a uniform. Flow seems to be really strict about this as a simple type cast between `Array<number>` and `[number, number, number]` is not allowed.

I tried quite a many different type permutations but couldn't find any better than this one. I'm quite sure there's a proper way doing this with flow that I'm not aware of (cc @mourner).

Reply:
thanks @mourner!